### PR TITLE
NAS-109314 / 21.04 / Save preferred trains for a user

### PR DESCRIPTION
**Pull request by sonicaj**
This PR introduces changes where we save preferred train for a user which UI will use to display catalog items. This helps the user to not have to each time go and change/select train for a catalog in the UI for installing some application.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/catalogs_linux/update.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class CatalogModel(sa.Model):
     repository = sa.Column(sa.Text(), nullable=False)
     branch = sa.Column(sa.String(255), nullable=False)
     builtin = sa.Column(sa.Boolean(), nullable=False, default=False)
+    preferred_trains = sa.Column(sa.JSON(type=list))
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/catalogs_linux/update.py:25]: Why don't we make it `nullable=False`?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/catalogs_linux/update.py (hunk 1 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/catalogs_linux/update.py:25]: It's `nullable=False` by default

Comment (themylogin) [reply src/middlewared/middlewared/plugins/catalogs_linux/update.py:25]: Then your migration lacks code to set NULL column to NOT NULL

**Review by themylogin — APPROVED** [review]
Thank you!

**Review by william-gr — APPROVED** [review]

**Pull request merged**